assets/js/controllers/sfox/sfoxVerify.controller.js

```diff
@@ -2,10 +2,20 @@ angular
   .module('walletApp')
   .controller('SfoxVerifyController', SfoxVerifyController);
 
-function SfoxVerifyController ($rootScope, $scope, $q, state, $http, sfox, modals, Upload, QA) {
-  $scope.states = state.stateCodes;
+function SfoxVerifyController ($rootScope, $scope, $q, state, $http, sfox, modals, Upload, QA, Options) {
   let exchange = $scope.vm.exchange;
 
+  let processOptions = (options) => {
+    let states = options.partners.sfox.states;
+    $scope.states = state.stateCodes.filter((s) => states.indexOf(s.Code) > -1);
+  };
+
+  if (Options.didFetch) {
+    processOptions(Options.options);
+  } else {
+    Options.get().then(processOptions);
```

Comment:
Looks like that was already done here: https://github.com/blockchain/My-Wallet-V3-Frontend/blob/master/assets/js/services/modals.service.js#L43-L44